src/classes/ADDR_Contact_TEST.cls

```diff
@@ -0,0 +1,1112 @@
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        Test.startTest();
+        insert addrs;
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;        
+        Test.stopTest();
+
+        // verify Contact address is new
+        map<Id, Contact> mapAccIdAcc = new map<Id, Contact>([select Id, Name, MailingStreet, MailingCity, MailingState, 
+        MailingPostalCode, MailingCountry, MailingLatitude, MailingLongitude, is_Address_Override__c, Current_Address__c from Contact]);
+        
+        for (Contact contact : mapAccIdAcc.values()) {
+            system.assertEquals('New Default Street\r\nSecond Line', contact.MailingStreet);
+            system.assertEquals('New Default Street\r\nSecond Line', contact.MailingStreet);
+        }
+        
+    }
+    
+    /*********************************************************************************************************
+    @description Update Contacts' Mailing address with multiline street address  
+    verify:
+        new Default Address created
+        Contacts get new Address
+        new Addresss has multiline street split into Street and Street2 fields
+    **********************************************************************************************************/            
+    @isTest
+    public static void updateMultilineStreetAddr() {
+        configSettings();
+         
+        // this creates a default Address for each Contact
+        UTIL_UnitTestData_TEST.ContactsWithAddrs contactsAddrs = UTIL_UnitTestData_TEST.createTestContactsAddrs(2);
+
+        // modify some of the Contact addresses directly
+        for (Integer i = 0; i < contactsAddrs.contacts.size(); i++) {
+            Contact contact = contactsAddrs.contacts[i];
+            contact.MailingStreet = 'Direct Street Edit\r\nSecond Line';
+            contact.MailingCity = 'Direct City Edit';
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        Test.startTest();
+        update contactsAddrs.contacts;
+        Test.stopTest();
+        
+        // verify Contact address
+        map<Id, Contact> mapAccIdAcc = new map<Id, Contact>([select Id, Name, MailingStreet, MailingCity, MailingState, 
+        MailingPostalCode, MailingCountry, MailingLatitude, MailingLongitude, is_Address_Override__c, Current_Address__c from Contact]);
+        
+        for (Contact contact : mapAccIdAcc.values()) {
+            system.assertEquals('Direct Street Edit\r\nSecond Line', contact.MailingStreet);
+            system.assertEquals('Direct Street Edit\r\nSecond Line', contact.MailingStreet);
+            system.assertEquals(false, contact.is_Address_Override__c);
+            system.assertNotEquals(null, contact.Current_Address__c);
+        }
+ 
+        // verify the Address objects split street.
+        List<Address__c> ListAddr = [select Id, MailingStreet__c, MailingStreet2__c, Formula_MailingStreetAddress__c, Default_Address__c from Address__c];
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. Consider either introducing a `WHERE` clause or contain the number of records redurned, or adding a `LIMIT` clause to cap the maximum number of records returned.